### PR TITLE
python3packages.sip: 4.19.24 -> 5.5.0.dev2010041444 

### PR DESCRIPTION
**Pull request by FRidh**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/sip/default.nix [hunk 1]
```diff
@@ -1,44 +1,41 @@
-{ lib, fetchurl, buildPythonPackage, python, isPyPy, sip-module ? "sip" }:
+{ lib
+, fetchurl
+, buildPythonPackage
+, isPyPy
+, packaging
+, toml
+, pythonOlder
+, python
```
Comment (FRidh) [inline pkgs/development/python-modules/sip/default.nix:8]: unused `python`

**Comment by FRidh** [comment]
The pyqt5 configure and build instructions are not really up to date. They say

> Prior to the release of SIP v5 the only way to build PyQt5 (and related projects) was based on a configure.py script. This method is now deprecated and will be removed when SIP v6 is released (which is expected to be mid-2020).

however, don't offer any other method (the configuration file needs to be passed to `configure.py`.

**Comment by eduardosm** [comment]
@FRidh  What's the state of this?

sip5 is required to build calibre5. You can check [this branch](https://github.com/ento/nixpkgs/commits/calibre5-pyqt5-sip5), where sip5 is added as new package, while sip stays at version 4, and calibre5 builds successfully.

cc @ento

**Comment by FRidh** [comment]
SIp 5 indeed seems much simpler to install.

> You can check this branch, where sip5 is added as new package, while sip stays at version 4, and calibre5 builds successfully.

We cannot have multiple versions of sip in `python-packages.nix`, because it may cause conflicts. It is however fine if the sip 5 package for now is private to calibre.

**Comment by eduardosm** [comment]
> We cannot have multiple versions of sip in `python-packages.nix`, because it may cause conflicts.

Is that true? It looks like at least Arch, Debian and Fedora includes both, which makes me think that sip5 is not backwards compatible with sip4 (can be checked quickly at https://pkgs.org/).

> It is however fine if the sip 5 package for now is private to calibre.

How would that be made? Note that calibre also needs pyqt5 to use sip5.

**Comment by FRidh** [comment]
> How would that be made? Note that calibre also needs pyqt5 to use sip5.

Ah, ok. Yes, we really need to get that working then. Maybe the pyqt5 build is actually simpler now as well, I just started off with the old one to ensure we did not lose anything. If the tarballs on PyPI and riverbank site are the same, then that's good.

**Comment by eduardosm** [comment]
I have created https://github.com/NixOS/nixpkgs/pull/108041, where sip5 is added as an additional package. Right now I'm running some nixpkgs-review iterations.

**Comment by mexisme** [comment]
> > We cannot have multiple versions of sip in `python-packages.nix`, because it may cause conflicts.
> 
> Is that true? It looks like at least Arch, Debian and Fedora includes both, which makes me think that sip5 is not backwards compatible with sip4 (can be checked quickly at https://pkgs.org/).

Does it matter that sip4 is no longer supported by upstream?
I feel like an unsupported version should not be the default, though.

**Comment by sbourdeauducq** [comment]
See also: https://github.com/NixOS/nixpkgs/issues/74974#issuecomment-704824940

**Pull request closed without merging**